File: backend/ride-share/src/main/java/mk/ukim/finki/rideshare/web/response/BookingResponse.java
package mk.ukim.finki.rideshare.web.response;

public record BookingResponse(
        Long id,
        String bookedOnDate,
        String bookedOnTime,
        String statusName,
        String statusPrettyName,
        Long bookedById,
        String bookedByUsername,
        RideResponse ride
) {
}

File: frontend/ride-share/src/interfaces/response/BookingResponse.ts
import { RideResponse } from "./RideResponse";

export interface BookingResponse {
    id: number;
    bookedOnDate: string;
    bookedOnTime: string;
    statusName: string;
    statusPrettyName: string;
    bookedById: number;
    bookedByUsername: string;
    ride: RideResponse;
}